evap/staff/tests/test_forms.py

```diff
@@ -719,3 +762,48 @@ def test_unused_questionnaire_visibility(self):
 
         form = EvaluationForm(instance=evaluation, semester=evaluation.course.semester)
         self.assertIn(questionnaire, form.fields["general_questionnaires"].queryset)
+
+
+class EvaluationCopyFormTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.semester = baker.make(Semester)
+        cls.course = baker.make(Course, semester=cls.semester)
+        cls.evaluation = baker.make(
+            Evaluation, course=cls.course, name_de="Das Original", name_en="The Original",
+            last_modified_time=datetime(2020, 1, 1), last_modified_user=baker.make(UserProfile))
+        cls.participants = baker.make(UserProfile, _quantity=8)
+        cls.evaluation.participants.set(cls.participants)
+        cls.evaluation.voters.set(cls.participants[:6])
+        cls.general_questionnaires = baker.make(Questionnaire, _quantity=5)
+        cls.evaluation.general_contribution.questionnaires.set(cls.general_questionnaires)
+
+    def test_initial_from_original(self):
+        form = EvaluationCopyForm(None, self.evaluation)
+        self.assertEqual(form['course'].initial, self.course.pk)
+        self.assertCountEqual(form.fields['course'].queryset, self.semester.courses.all())
+        self.assertEqual(form['name_de'].initial, "Das Original")
+        self.assertEqual(form['name_en'].initial, "The Original")
+        self.assertCountEqual(form['participants'].initial, self.participants)
+        self.assertGreater(form['last_modified_time'].initial, self.evaluation.last_modified_time)
+        self.assertEqual(form['last_modified_user_name'].initial, None)
+        self.assertCountEqual(form['general_questionnaires'].initial, self.general_questionnaires)
```

Comment:
I tried to write this test as view test, but this was very inconvenient, as I had to stringify nearly everything :/
But, there is currently no guarantee that the `EvaluationCopyForm` will be used by the form – using `response.context['evaluation_form']` would combine both benefits (only thing that is not checked whether the form is rendered correctly, but I think this is django’s job)
What do you think?